crates/cranelift/src/debug/write_debuginfo.rs

```diff
@@ -146,7 +149,14 @@ fn create_frame_table<'a>(
     let cie_id = table.add_cie(isa.create_systemv_cie()?);
 
     for (i, metadata) in funcs {
+        let mut unwind_info: Option<&CfaUnwindInfo> = None;
         if let Some(UnwindInfo::SystemV(info)) = &metadata.unwind_info {
+            unwind_info = Some(info);
+        } else if let Some(info) = &metadata.cfa_unwind_info {
+            unwind_info = Some(info);
+        }
```

Comment:
This makes me pause for a bit for a few reasons. One is could this assert that at most one of these fields are `SystemV` unwind information? Hopefully to head off any possible future confusion.

The second reason is that in Wasmtime [only `unwind_info` is consulted](https://github.com/bytecodealliance/wasmtime/blob/9377dfd7b82c8cc5bfdfbc435df1b633b2506f08/crates/cranelift-shared/src/obj.rs#L132-L134) meaning that it seems to be like we wouldn't emit the DWARF unwinding information? It also seems though like you're probably running/testing on Windows, so I'm curious if you are how this is working?

Reply:
> Could this assert that at most one of these fields are SystemV unwind information?

Sure, will add. It's not really _incorrect_ for them to be `{SysV, SysV}`, just wasteful in terms of memory.

> I'm curious how this is working?

This function is a callee of `append_dwarf`, which itself adds to the final image. So on Windows we will have `.debug_frame`, much like the other DWARF sections.

It is interesting to note what happens when we are on Unix - in that case the unwind info will be effectively emitted twice (note this is pre-existing behavior). AFAICT from the code the "unwind" part will be emitted into `.eh_frame`, so they don't conflict.

In effect, the "debug info" use of this information is completely decoupled from the "unwind" use.